src/mixins/ui/lasso-tool-ui.js

```diff
@@ -65,6 +168,10 @@ class ShapeHandler {
     shape.registerEvents(Object.values(LassoShapeEventConstants))
     this.drawEngine.addShape(shape, selectOpts)
     this.drawEngine.moveShapeToTop(shape)
+
+    // now tag the shape to the originating tool, which may be useful
+    // for callback handlers to know
+    shape.lasso_tool_type = this.getLassoToolType()
```

Comment:
tags created shapes with the lasso tool type too, in case that's useful. again, would be great to document this somewhere .... but alas.